src/components/Panels/Telescope/Targets/TargetSwapButton.tsx

```diff
@@ -0,0 +1,98 @@
+import { useCanEdit } from '@/components/atoms/auth';
+import {
+  Instrument as InstrumentName,
+  InstrumentSpecificsInput,
+  RotatorTrackingInput,
+  TargetPropertiesInput,
+} from '@gql/server/gen/graphql';
+import { useNavigateState } from '@gql/server/NavigateState';
+import { useRestoreTarget, useSwapTarget } from '@gql/server/TargetSwap';
+import { Button } from 'primereact/button';
+import { Target } from '@gql/configs/gen/graphql';
+import { useInstrument } from '@gql/configs/Instrument';
+import { useRotator } from '@gql/configs/Rotator';
+import { useToast } from '@/Helpers/toast';
+
+export function TargetSwapButton({ selectedTarget }: { selectedTarget: Target | undefined }) {
+  const canEdit = useCanEdit();
+  const toast = useToast();
+
+  const { data, loading: stateLoading } = useNavigateState();
+  const [swapTarget, { loading: swapLoading }] = useSwapTarget();
+  const [restoreTarget, { loading: restoreLoading }] = useRestoreTarget();
+
+  const { data: instrumentData, loading: instrumentLoading } = useInstrument();
```

Comment:
I think this will cause the same problem we had in the last Friday tests, where instruments was always undefined, because we are not passing the required variables `name`, `issPort` and `wfs`.